android/src/main/java/com/alinz/parkerdan/shareextension/ShareModule.java

```diff
@@ -53,23 +68,23 @@ public WritableMap processIntent() {
         if (type == null) {
           type = "";
         }
+
         if (Intent.ACTION_SEND.equals(action) && "text/plain".equals(type)) {
-          value = intent.getStringExtra(Intent.EXTRA_TEXT);
-        }
-        else if (Intent.ACTION_SEND.equals(action) && ("image/*".equals(type) || "image/jpeg".equals(type) || "image/png".equals(type) || "image/jpg".equals(type) ) ) {
+          text = intent.getStringExtra(Intent.EXTRA_TEXT);
+        } else if (Intent.ACTION_SEND.equals(action) && type.startsWith("image")) {
           Uri uri = (Uri) intent.getParcelableExtra(Intent.EXTRA_STREAM);
-         value = "file://" + RealPathUtil.getRealPathFromURI(currentActivity, uri);
-
-       } else {
-         value = "";
-       }
-      } else {
-        value = "";
-        type = "";
-      }
+          images.pushString("file://" + RealPathUtil.getRealPathFromURI(currentActivity, uri));
+        } else if (Intent.ACTION_SEND_MULTIPLE.equals(action) && type.startsWith("image")) {
```

Comment:
Why are images the only things that can be "mutiple"?  Is it not possible to send more than one thing of anther type?

Perhaps the shape of the object we return to RN is wrong.  Maybe it should be an array of {type, value}?  This way you could use it to share multiple contacts, image etc...